src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -317,6 +317,22 @@ internal static Syntax.InternalSyntax.DirectiveStack ApplyDirectives(this Syntax
             return builder.ToList();
         }
 
+        #region ITypeSymbol
+
+        /// <summary>
+        /// Returns true iff the supplied type is sbyte, byte, short, ushort, int, uint,
+        /// long, ulong, bool, char, string, or an enum-type, or if it is the nullable type
+        /// corresponding to one of those types. These types were permitted as the governing
+        /// type of a switch statement in C# 6.
+        /// </summary>
+        public static bool IsValidV6SwitchGoverningType(this ITypeSymbol type)
```

Comment:
Maybe I'm missing something, but it looks like the EnC project does not have access to compiler internals (and trying to give it access causes many conflicts). I'm assuming there is a reason for that.
If so, having this public compiler API will help.